### ISSUE TITLE
Support for --enable-relocatable

### ISSUE THREAD

**Issue by simonmar** [bgamari/build-env#25]
I'd really like `build-env` to support building packages with `--enable-relocatable`. Right now it fails with errors like this

```
build-env build async -f src --cabal cabal --configure-arg --enable-relocatable --prefix /tmp/exp/install --prefetched --ghc ghc --ghc-pkg ghc-pkg
Computing build plan

Building and installing units sequentially.
NB: pass -j<N> for increased parallelism.             
=== BUILD START ===       
Finished building lib:base-orphans
                                                               
 ############
 ## 1 of 4 ##                                                                                                                   ############

                               
Finished building lib:data-array-byte
Setup: Library directory of a dependency:
"/tmp/exp/src/base-orphans-0.9.2/temp-build/base-orphans-0.9.2-d21130a4df03e9f7007781a2a3a349c614f9f784a12cafce744dcdbef52dc9c5/build"                                                                                                                        
is not relative to the installation prefix:                                                                                    
"/tmp/exp/install"                                                                              
```

Currently we hack the package DB after running `build-env build` to replace the prefix with `${pkgroot}`, but this doesn't help with binaries that still look in prefix to find data files.

**Comment by simonmar** [comment]
That certainly helps. The issue with executables is that by default `getDataDir` returns the absolute `datadir` that was specified at compile-time (it can be overriden with an environment variable). If you specify `--enable-relocatable` to Cabal then `getDataDir` figures out the datadir relative to the executable's location, so it works if you move the whole installation somewhere else. (unfortunately this behaviour is currently enabled only for Windows for reasons I don't remember, but it's a 1-line patch to Cabal to enable it on Linux too).

**Comment by sheaf** [comment]
This should be fixed by 8990cff. I've added a test which uses `--enable-relocatable`, and then relocates and runs `alex`; it manages to find its data files. (This seems to go against what you said that this behaviour isn't enabled on Linux?)

**Comment by simonmar** [comment]
Here's why I think it shouldn't work on Linux: https://github.com/haskell/cabal/blob/9c775f2f6dd9ce03f62573f40e2738487cbe0c2b/Cabal/src/Distribution/Simple/Build/PathsModule.hs#L74-L87
